common/scala/src/main/scala/org/apache/openwhisk/core/entity/Parameter.scala

```diff
@@ -144,19 +151,19 @@ protected[entity] class ParameterName protected[entity] (val name: String) exten
  * be undefined, such as when an action is created but the parameter value is not bound yet.
  * In general, this is an opaque value.
  *
- * It is a value type (hence == is .equals, immutable and cannot be assigned null).
  * The constructor is private so that argument requirements are checked and normalized
  * before creating a new instance.
  *
- * @param value the value of the parameter, may be null
+ * @param v the value of the parameter, may be null
+ * @param init if true, this parameter value is only offered to the action during initialization
  */
-protected[entity] class ParameterValue protected[entity] (private val v: JsValue) extends AnyVal {
+protected[entity] case class ParameterValue protected[entity] (private val v: JsValue, val init: Boolean) {
 
   /** @return JsValue if defined else JsNull. */
-  protected[entity] def value = Option(v) getOrElse JsNull
+  protected[entity] val value = Option(v) getOrElse JsNull
```

Comment:
Just curious - Any reason to make them `val` instead of `def` as they would then consume small memory